spec/requests/bookmarks_controller_spec.rb

```diff
@@ -142,4 +144,62 @@
       end
     end
   end
+
+  describe "#bulk" do
+    it "needs you to be logged in" do
+      put "/bookmarks/bulk.json"
+      expect(response.status).to eq(403)
+    end
+
+    describe "when logged in" do
+      before { sign_in(current_user) }
+
+      let!(:operation) { { type: "clear_reminder" } }
+      let!(:bookmark_ids) { [1, 2] }
+
+      it "requires a list of bookmark_ids" do
+        put "/bookmarks/bulk.json", params: { operation: operation }
+        expect(response.status).to eq(400)
+      end
+
+      it "requires an operation param" do
+        put "/bookmarks/bulk.json", params: { bookmark_ids: bookmark_ids }
+        expect(response.status).to eq(400)
+      end
+
+      it "can clear reminder for the given bookmarks" do
+        Guardian.any_instance.stubs(:can_edit?).returns(true)
+        bookmark = Fabricate(:bookmark_next_business_day_reminder)
+
+        expect do
+          put "/bookmarks/bulk.json",
+              params: {
+                operation: {
+                  type: "clear_reminder",
+                },
+                bookmark_ids: [bookmark.id],
+              }
+          expect(response.status).to eq(200)
+        end.to change { Bookmark.find(bookmark.id).reminder_set_at }.to(nil)
+      end
+
+      it "can delete bookmarks" do
+        Guardian.any_instance.stubs(:can_edit?).returns(true)
```

Comment:
Same here, given this is a destructire operation, we want to make sure the only right users can do this.